datahub-web-react/src/app/shared/handleGraphQLError.ts

```diff
@@ -0,0 +1,40 @@
+import { ErrorResponse } from '@apollo/client/link/error';
+import { message } from 'antd';
+import { ErrorCodes } from './constants';
+
+interface Props {
+    error: ErrorResponse;
+    defaultMessage: string;
+    permissionMessage?: string;
+    badRequestMessage?: string;
+    serverErrorMessage?: string;
+}
+
+export default function handleGraphQLError({
+    error,
+    permissionMessage,
+    defaultMessage,
+    badRequestMessage,
+    serverErrorMessage,
+}: Props) {
+    // destroy the default error message from errorLink in App.tsx
+    message.destroy();
+    const { graphQLErrors } = error;
+    if (graphQLErrors && graphQLErrors.length) {
+        const { extensions } = graphQLErrors[0];
+        const errorCode = extensions && (extensions.code as number);
+        if (errorCode === ErrorCodes.Forbidden && permissionMessage) {
+            message.error(permissionMessage);
+            return;
+        }
```

Comment:
This should always return a permission error message (generic) if that's what's caused the failure.